File: src/1-500/011/MostWater.py
from typing import List


class Solution:
    def maxArea(self, height: List[int]) -> int:
        ans = 0
        l = 0
        r = len(height) - 1
        while l < r:
            ans = max(ans, min(height[l], height[r]) * (r - l))
            if height[l] < height[r]:
                l += 1
            else:
                l -= 1
        return ans


solution = Solution()
print(solution.maxArea([1,2,4,3]))

File: src/1-500/011/MostWater.go
package main

func min(a, b int) int {
	if a <= b {
		return a
	}
	return b
}

func max(a, b int) int {
	if a >= b {
		return a
	}
	return b
}

func maxArea(height []int) int {
	right := len(height) - 1
	left := 0
	area := 0
	for left < right {
		area = max(area, (right-left)*min(height[left], height[right]))
		if height[right] > height[left] {
			left++
		} else {
			right--
		}
	}
	return area
}

func main() {
	///fmt.Print(lengthOfLongestSubstring("abcabc"))
}
